### PR TITLE
Optimize inmemory index cache WithLabelValues call

### PR DESCRIPTION
**Pull request by yeya24**
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

image

From CPU profile of store gateway, inmemory index cache `get` method took about 13% of CPU time, while `CounterVec.WithLabelValues` took about 4% of CPU. Now we call `WithLabelValues` every time when we call `get` method and it can be expensive if we fetch a large amount of series/potings. 

This can be optimized by only update counter once at the end of `FetchMultiSeries` and `FetchMultiPostings` method. 

## Verification

Existing tests should still pass. The change doesn't affect existing behavior.

### CODE REVIEW

**Comment by yeya24** [comment]
Anyone wants to take a look? Maybe @saswatamcode or @MichaHoffmann?

**Review by saswatamcode — APPROVED** [review]
LGTM!

File: pkg/store/cache/inmemory.go [hunk 1]
```diff
@@ -360,17 +367,25 @@ func (c *InMemoryIndexCache) FetchMultiSeries(ctx context.Context, blockID ulid.
...
+		requests++
+		if b, ok := c.get(cacheKey{blockIDKey, cacheKeySeries(id), ""}); ok {
+			hit++
 			hits[id] = b
 			continue
 		}
 
 		misses = append(misses, id)
 	}
+	c.commonMetrics.requestTotal.WithLabelValues(cacheTypeSeries, tenant).Add(float64(requests))
```
Comment (fpetkovski) [inline pkg/store/cache/inmemory.go:387]: Does it make sense to call `WithLabelValues` at the start and store it into a variable that we can use everywhere?

Comment (yeya24) [reply pkg/store/cache/inmemory.go:387]: Ah I see what you mean. You mean we call `WithLabelValues` once, store it as a var and then we can call its method multiple times.

I am okay with it, but I don't see much difference between it and my current approach. 
They both call `WithLabelValues` once, but one approach calls `Inc` multiple times, another approach calls `Add` once.
Performance sounds similar as main bottleneck is `WithLabelValues`

**Comment by yeya24** [comment]
Thanks for the reviews!

**Pull request merged**